akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -572,6 +572,16 @@ object Logging {
   }
 
   /**
+    * Obtain LoggingAdapter with MDC support for the given actor.
+    * Don't use it outside its specific Actor as it isn't thread safe
+    */
+  def getLogger(logSource: Actor): DiagnosticLoggingAdapter = {
+    val (str, clazz) = LogSource.fromAnyRef(logSource)
+    val system = logSource.context.system.asInstanceOf[ExtendedActorSystem]
+    new BusLogging(system.eventStream, str, clazz, system.logFilter) with DiagnosticLoggingAdapter
+  }
```

Comment:
can this delegate to `def apply(logSource: Actor): DiagnosticLoggingAdapter` instead?